### PR TITLE
Mp add authorization

### PR DESCRIPTION
**Pull request by Ritumka**

### CODE REVIEW

**Review by tluu5 — COMMENTED** [review]
great job adding before_action on the photos_controller

**Review by tluu5 — COMMENTED** [review]
I like how you wrote the if statement for the current_user, allowing only the user to see the show page.

**Review by tluu5 — APPROVED** [review]

File: config/routes.rb
Comment (merrillhuang) [inline config/routes.rb:1]: Good job excluding comments index route.

File: app/controllers/comments_controller.rb
Comment (merrillhuang) [inline app/controllers/comments_controller.rb:1]: A user can still edit and delete comments that don't belong to them. If I visit comments/4/edit as Alice, I can still edit the comment even though it belongs to Aubrey.

File: app/controllers/follow_requests_controller.rb
Comment (merrillhuang) [inline app/controllers/follow_requests_controller.rb:1]: Users can't accept or reject follow requests when they are recipients. Check /manda when logged in as Alice and try accepting or rejecting the follow request.

**Review by merrillhuang — CHANGES_REQUESTED** [review]

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -58,13 +60,28 @@ def destroy
...
+    if current_user != @photo.owner && @photo.owner.private? && !current_user.leaders.include?(@photo.owner)
+      redirect_back fallback_location: root_url, alert: "Not authorized"
     end
+  end
 
-    # Only allow a list of trusted parameters through.
-    def comment_params
-      params.require(:comment).permit(:author_id, :photo_id, :body)
+  def is_a_comment_author
+    @comment = Comment.find(params.fetch(:id))
```
Comment (marywebby) [inline app/controllers/comments_controller.rb:77]: good job implementing the delete comment section here.

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -1,5 +1,7 @@
 class CommentsController < ApplicationController
   before_action :set_comment, only: %i[ show edit update destroy ]
+  before_action :is_an_authorized_user, only: [:create]
+  before_action :is_a_comment_author, only: [:destroy]
```
Comment (marywebby) [inline app/controllers/comments_controller.rb]: it looks like edit is still working here as well, but maybe make sure to include `:edit` on the `:is_a_comment_author`

File: app/controllers/follow_requests_controller.rb [hunk 3]
```diff
@@ -1,5 +1,6 @@
 class FollowRequestsController < ApplicationController
```
Comment (marywebby) [inline app/controllers/follow_requests_controller.rb:1]: Also agree with what Merrill said, the user cannot accept or reject the pending follow requests for them.

File: app/controllers/likes_controller.rb [hunk 4]
```diff
@@ -1,5 +1,6 @@
 class LikesController < ApplicationController
   before_action :set_like, only: %i[ show edit update destroy ]
+  before_action :ensure_current_user_is_owner, only: [:destroy, :update, :edit]
```
Comment (marywebby) [inline app/controllers/likes_controller.rb:3]: Good job implementing likes

File: app/controllers/photos_controller.rb [hunk 5]
```diff
@@ -50,11 +51,15 @@ def update
...
   # DELETE /photos/1 or /photos/1.json
   def destroy
+    if current_user == @photo.owner
     @photo.destroy
     respond_to do |format|
       format.html { redirect_back fallback_location: root_url, notice: "Photo was successfully destroyed." }
       format.json { head :no_content }
     end
+  else
+    redirect_back(fallback_location: root_url, notice: "Nice try, but that is not your photo.")
```
Comment (marywebby) [inline app/controllers/photos_controller.rb:61]: very funny

**Review by marywebby — CHANGES_REQUESTED** [review]
Great job!! Please fix the accepting and rejecting functionality of the follow requests, and re-request me when you're done!

File: app/controllers/follow_requests_controller.rb (hunk 3 above)
Comment (Ritumka) [reply app/controllers/follow_requests_controller.rb:1]: Fixed!

File: app/controllers/follow_requests_controller.rb
Comment (Ritumka) [reply app/controllers/follow_requests_controller.rb:1]: Fixed!

**Review by merrillhuang — APPROVED** [review]
Looks good!

**Review by marywebby — APPROVED** [review]
Great work!!!

**Pull request merged**